### PR TITLE
Refactor repo-legacy.js, remove messy global variables. Fix errors.

### PR DESCRIPTION
**Pull request by wxiaoguang**
The JS global variable `commentMDEditors` and `autoSimpleMDE` are removed. JS code uses `$('textarea').data('simplemde')` directly.

Two JS errors are also fixed:
* Fix an error in Sortable
* Fix an incorrect call assignMenuAttributes from the template

And the nodejs version mentioned in document is too old, so I just changed it to `Node.js LTS`  :)

### CODE REVIEW

File: templates/repo/diff/comment_form.tmpl [hunk 1]
```diff
@@ -9,7 +9,7 @@
 		<input type="hidden" name="diff_start_cid">
 		<input type="hidden" name="diff_end_cid">
 		<input type="hidden" name="diff_base_cid">
-		<div class="ui top tabular menu" {{if not $.hidden}}onload="assignMenuAttributes(this)" {{end}}data-write="write" data-preview="preview">
```
Comment (silverwind) [inline templates/repo/diff/comment_form.tmpl:12]: `assignMenuAttributes` becomes unused with this, should it be deleted?

File: web_src/js/features/repo-legacy.js [hunk 2]
```diff
@@ -259,6 +254,171 @@ export function initRepoCommentForm() {
...
+    $textarea.val($rawContent.text());
+    $simplemde.value($rawContent.text());
+  }
+  requestAnimationFrame(() => {
+    $textarea.focus();
+    $simplemde.codemirror.focus();
+  });
+}
+
+
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
```

File: web_src/js/features/repo-legacy.js [hunk 3]
```diff
@@ -523,55 +526,57 @@ export function initRepository() {
...
-  initRepoSettingBranches();
-  initRepoCommonLanguageStats();
 }
 
-function initRepoIssueQuoteReply() {
+function initRepoIssueCommentEdit() {
+  // Issue/PR Context Menus
+  $('.comment-header-right .context-dropdown').dropdown({
+    action: 'hide',
+  });
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
  $('.comment-header-right .context-dropdown').dropdown({action: 'hide'});
```

File: web_src/js/features/repo-legacy.js [hunk 4]
```diff
@@ -259,6 +254,171 @@ export function initRepoCommentForm() {
...
+          $renderContent.html(data.content);
+          $rawContent.text($textarea.val());
+        }
+        const $content = $segment;
+        if (!$content.find('.dropzone-attachments').length) {
+          if (data.attachments !== '') {
+            $content.append(`
+                  <div class="dropzone-attachments">
+                  </div>
+                `);
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
            $content.append(`<div class="dropzone-attachments"></div>`);
```

File: web_src/js/features/repo-legacy.js [hunk 5]
```diff
@@ -259,6 +254,171 @@ export function initRepoCommentForm() {
...
+              $dropzone.find('.files').empty();
+              $.each(data, function () {
+                const imgSrc = `${$dropzone.data('link-url')}/${this.uuid}`;
+                dz.emit('addedfile', this);
+                dz.emit('thumbnail', this, imgSrc);
+                dz.emit('complete', this);
+                dz.files.push(this);
+                fileUuidDict[this.uuid] = {
+                  submitted: true,
+                };
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
                fileUuidDict[this.uuid] = {submitted: true};
```

File: web_src/js/features/repo-legacy.js [hunk 6]
```diff
@@ -259,6 +254,171 @@ export function initRepoCommentForm() {
...
+        dictRemoveFile: $dropzone.data('remove-file'),
+        timeout: 0,
+        thumbnailMethod: 'contain',
+        thumbnailWidth: 480,
+        thumbnailHeight: 480,
+        init() {
+          this.on('success', (file, data) => {
+            fileUuidDict[file.uuid] = {
+              submitted: false,
+            };
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
            fileUuidDict[file.uuid] = {submitted: false};
```

**Comment by silverwind** [comment]
Good refactor, these globals were a very ugly hack.

File: templates/repo/diff/comment_form.tmpl (hunk 1 above)
Comment (wxiaoguang) [reply templates/repo/diff/comment_form.tmpl:12]: Nope, there are other callers.

image

**Comment by wxiaoguang** [comment]
The legacy code is still very difficult to understand, so we have to continue to refactor them again and again.

ps: during refactoring, usually I just do copy and paste to untouched code and keep most code as before.

**Comment by silverwind** [comment]
I guess this editor stuff should ideally be rewritten as a Vue (or dare I say React) component. Having such code in declarative form will always be a pain.

**Comment by wxiaoguang** [comment]
I agree to introduce some MVVM mechanism. Since we are still using these jQuery code heavliy, I think it's good to make them clear first. And if the legacy code becomes clear, it's easier to migrate them to new frameworks. That's why I am doing some refactoring work.

**Review by Gusted — APPROVED** [review]
LGTM

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Code looks good but I haven't gotten around to testing this.

**Comment by wxiaoguang** [comment]
> Code looks good but I haven't gotten around to testing this.

I have tested it on my side. If there is any bug, I will fix it ASAP.

**Pull request merged**